src/averageAges.js

```diff
@@ -54,8 +72,29 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+function findMother(people, child) {
+  return people.some(mother => mother.name === child.mother);
+}
+
+function motherBorn(people, mother) {
+  const mom = people.find(person => person.name === mother);
+
+  return mom.born;
+}
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(person => (
+    onlyWithSon
+      ? person.sex === 'm' && findMother(people, person)
+      : findMother(people, person)
+  ));
+
+  const diff = child.map(kid => kid.born - motherBorn(people, kid.mother));
```

Comment:
```suggestion
  const ageDifferences = child.map(kid => kid.born - motherBorn(people, kid.mother));
```
We name array in plural form